Replicator/IncomingRev.cc

```diff
@@ -359,6 +359,10 @@ namespace litecore { namespace repl {
             }
         }
 
+        if (!(_rev->flags & kRevHasAttachments) && _db->hasBlobReferences(root)) {
+            _rev->flags |= kRevHasAttachments;
+        }
```

Comment:
As commented in the slack, I think we should reuse the logic in line 330 - 343, as we don't need to find blob for the case that delta sync is not used AND no blobs changes.

The change could be : 

```
if (_mayContainBlobChanges || <didApplyDelta>) {
            _db->findBlobReferences(root, true, [=](FLDeepIterator i, Dict blob, const C4BlobKey &key) {
                _rev->flags |= kRevHasAttachments;
                if (_mayContainBlobChanges) {
                   _pendingBlobs.push_back({_rev->docID,
                                         alloc_slice(FLDeepIterator_GetPathString(i)),
                                         key,
                                         blob["length"_sl].asUnsigned(),
                                         C4Blob::isLikelyCompressible(blob)});
                   _blob = _pendingBlobs.begin();
                }
            });
}
```